### ISSUE TITLE
Include useful diagnostic information in errors

### ISSUE THREAD

**Issue by fitzgen** [bytecodealliance/regalloc2#147]
Right now, we just give error messages like `BB(Block(0))`, meaning that there is some error somewhere in block 0.

If one enables logging, then one gets messages like "block 0 does not end in a terminator" or something like that.

It would be nice if we put that information in the actual error itself, so folks didn't have to enable logging to figure out what went wrong.

**Comment by cfallin** [comment]
Agreed -- this would be an easy starter PR if someone wants to take it on!
